core/main.go

```diff
@@ -61,22 +74,42 @@ func deployCoreServices(c *config.Config, api *api.API) error {
 	return nil
 }
 
+func stopRunningServices(api *api.API) error {
+	services, err := api.ListServices()
+	if err != nil {
+		return err
+	}
+	for _, s := range services {
+		status, err := s.Status()
+		if err != nil {
+			return err
+		}
+		if status == service.STARTING || status == service.PARTIAL || status == service.RUNNING {
+			if err := api.StopService(s.Hash); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
 func main() {
-	// init configs.
-	c, err := config.Global()
+	dep, err := initDependencies()
 	if err != nil {
-		log.Fatal(err)
+		logrus.Fatalln(err)
 	}
 
 	// init logger.
-	logger.Init(c.Log.Format, c.Log.Level, c.Log.ForceColors)
+	logger.Init(dep.config.Log.Format, dep.config.Log.Level, dep.config.Log.ForceColors)
 
-	// init gRPC server.
-	server, err := initGRPCServer(c)
-	if err != nil {
+	// init system services.
+	if err := deployCoreServices(dep.config, dep.api); err != nil {
```

Comment:
I think it make more sense to start the gRPC server before deploying the Core services:
The services will fail (and restart) until the gRPC server is up and running.

Reply:
just kept the same behavior here as before just refactors it but the behavior is the same.
I agree it's not perfect, optimal solution would be:
- deploy service
- start service api
- start deployed services
- start core api

But this is not the purpose of this PR